From 6434b6f2b901b376ba2efa5652b76fc68f5e75e0 Mon Sep 17 00:00:00 2001
From: Michael Holzheu <holzheu@linux.vnet.ibm.com>
Date: Thu, 17 Jan 2019 17:52:15 +0100
Subject: [PATCH 1/2] lapi/regen.sh: Fix s390 check

On s390 (32 bit) only the "__s390__" macro is set by gcc and on
s390x (64 bit) both macros "__s390__" and "__s390x__" are set.

Therefore a check "#ifdef __s390__" is not sufficient for checking
32 bit Mainframe Linux systems since this is also true on s390x machines.

Fix this and generate the correct s390 check:

  "#if defined(__s390__) && !defined(__s390x__)"

Signed-off-by: Michael Holzheu <holzheu@linux.vnet.ibm.com>
Acked-by: Jan Stancek <jstancek@redhat.com>
---
 include/lapi/syscalls/regen.sh | 1 +
 1 file changed, 1 insertion(+)

diff --git a/include/lapi/syscalls/regen.sh b/include/lapi/syscalls/regen.sh
index e106dd3bd..51d612e2a 100755
--- a/include/lapi/syscalls/regen.sh
+++ b/include/lapi/syscalls/regen.sh
@@ -77,6 +77,7 @@ for arch in $(cat "${srcdir}/order") ; do
 	case ${arch} in
 		sparc64) echo "#if defined(__sparc__) && defined(__arch64__)" ;;
 		sparc) echo "#if defined(__sparc__) && !defined(__arch64__)" ;;
+		s390) echo "#if defined(__s390__) && !defined(__s390x__)" ;;
 		*) echo "#ifdef __${arch}__" ;;
 	esac
 	while read line ; do
-- 
2.19.1

